src/Microsoft.ML.Data/Transforms/ColumnConcatenatingEstimator.cs

```diff
@@ -116,8 +116,8 @@ private bool HasCategoricals(SchemaShape.Column col)
                 }
                 if (!col.ItemType.Equals(itemType))
                 {
-                    throw _host.Except($"Column '{sources[i]}' has values of {col.ItemType}, " +
-                        $"which is not the same as earlier observed type of {itemType}.");
+                    throw _host.Except($"Concatenated columns should have the same type. Column '{sources[i]}' has values of {col.ItemType}, " +
+                        $"but expected columns of type {itemType}.");
```

Comment:
"Column '{sources[i]}' has type of {col.ItemType}, " +
$"but expected column type is {itemType}.

Will that be more clear?

Reply:
Yeah, I can remove the first sentence, if we think the user will know concatenated columns must have the same type.

Reply:
I think the first sentence is allright. But maybe use the wording Frank suggested for the second one?

Reply:
Yes, first sentence is good, I mean change the second sentence little bit.

---
In reply to: [508886394](https://github.com/dotnet/machinelearning/pull/5444#discussion_r508886394) [](ancestors = 508886394)